core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```diff
@@ -58,13 +66,20 @@
    */
   protected static final String PATH_SEPARATOR = String.valueOf(PATH_SEPARATOR_CHAR);
 
+  /** Executor service used for parallel UFS operations such as bulk deletes. */
+  ExecutorService mExecutorService;
+
   /**
    * Constructs an {@link ObjectUnderFileSystem}.
    *
    * @param uri the {@link AlluxioURI} used to create this ufs
    */
   protected ObjectUnderFileSystem(AlluxioURI uri) {
     super(uri);
+
+    int numThreads = Configuration.getInt(PropertyKey.UNDERFS_OBJECT_STORE_SERVICE_THREADS);
+    mExecutorService = ExecutorServiceFactories.fixedThreadPoolExecutorServiceFactory(
+        "alluxio-underfs-object-service-worker", numThreads).create();
```

Comment:
Do the underlying client libraries already do some optimization for parallel bulk deletes?